apps/src/code-studio/headerShare.js

```diff
@@ -7,10 +7,25 @@ import ShareDialog from './components/ShareDialog';
 import {Provider} from 'react-redux';
 import {getStore} from '../redux';
 import {showShareDialog} from './components/shareDialogRedux';
+import {
+  setLibraryFunctions,
+  setLibraryName
+} from './components/libraryShareDialogRedux';
 import {AllPublishableProjectTypes} from '../util/sharedConstants';
+import experiments from '@cdo/apps/util/experiments';
 
 export function shareProject(shareUrl) {
   dashboard.project.saveIfSourcesChanged().then(() => {
+    const appType = dashboard.project.getStandaloneApp();
+    if (appType === 'applab' && experiments.isEnabled('student-libraries')) {
+      getStore().dispatch(
+        setLibraryFunctions(dashboard.project.getLibraryFromApp())
+      );
+      var projectName = dashboard.project.getCurrentName().replace(/\s+/g, '');
+      var libraryName =
```

Comment:
Would it cause problems if the project name had *just* symbols/numbers? (ex: "12" is a valid project name, but I could see `12.function()` causing problems..

Reply:
Great catch. I've fixed that now. Thank you!